src/lib/pass-provider.c

```diff
@@ -100,15 +100,30 @@ rnp_passphrase_provider_stdin(const pgp_passphrase_ctx_t *ctx,
     if (!ctx || !passphrase || !passphrase_size) {
         goto done;
     }
-    rnp_strhexdump(keyidhex, ctx->key->keyid, PGP_KEY_ID_SIZE, "");
-    snprintf(target, sizeof(target), "key 0x%s", keyidhex);
+
+    if ((ctx->op != PGP_OP_DECRYPT_SYM) && (ctx->op != PGP_OP_ENCRYPT_SYM)) {
+        rnp_strhexdump(keyidhex, ctx->key->keyid, PGP_KEY_ID_SIZE, "");
+        snprintf(target, sizeof(target), "key 0x%s", keyidhex);
+    }
 start:
-    snprintf(prompt, sizeof(prompt), "Enter passphrase for %s: ", target);
+    if (ctx->op == PGP_OP_DECRYPT_SYM) {
+        snprintf(prompt, sizeof(prompt), "Enter passphrase to decrypt data: ");
+    } else if (ctx->op == PGP_OP_ENCRYPT_SYM) {
```

Comment:
I love `switch` and `case` to this but it is a style question.

Reply:
Most likely when this get to more then 3 branches it will be converted to the switch :)